test/core/authorship/proposer_test.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "authorship/impl/proposer_impl.hpp"
+
+#include <gtest/gtest.h>
+#include "mock/core/authorship/block_builder_factory_mock.hpp"
+#include "mock/core/authorship/block_builder_mock.hpp"
+#include "mock/core/runtime/block_builder_api_mock.hpp"
+#include "mock/core/transaction_pool/transaction_pool_mock.hpp"
+#include "testutil/outcome.hpp"
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::Test;
+
+using kagome::authorship::BlockBuilder;
+using kagome::authorship::BlockBuilderFactoryMock;
+using kagome::authorship::BlockBuilderMock;
+using kagome::authorship::ProposerImpl;
+using kagome::common::Buffer;
+using kagome::primitives::Block;
+using kagome::primitives::BlockId;
+using kagome::primitives::BlockNumber;
+using kagome::primitives::Digest;
+using kagome::primitives::Extrinsic;
+using kagome::primitives::InherentData;
+using kagome::primitives::InherentIdentifier;
+using kagome::primitives::Transaction;
+using kagome::runtime::BlockBuilderApiMock;
+using kagome::transaction_pool::TransactionPoolMock;
```

Comment:
in tests it is better to use 
```
using namespace kagome;
using namespace primitivs;
// etc
```

Reply:
I hope this is not a rule )

Reply:
I don't like this approach. I am afraid there will be included something that is not needed. I don't see big trouble to include explicitly only what is needed

Reply:
This is just a suggetion, so you "can" use this "if you want"